Source/WTF/wtf/text/StringCommon.h

```diff
@@ -466,37 +549,107 @@ ALWAYS_INLINE const uint16_t* memchr16(const uint16_t* pointer, uint16_t charact
         return nullptr;
 
     ASSERT(index < length);
-    return memchr16AlignedImpl(pointer + index, character, length - index);
+    return find64AlignedImpl(pointer + index, character, length - index);
+}
+#else
+ALWAYS_INLINE const uint64_t* find64(const uint64_t* pointer, uint64_t character, size_t length)
+{
+    for (size_t index = 0; index < length; ++index) {
+        if (pointer[index] == character)
+            return pointer + index;
+    }
+    return nullptr;
+}
+#endif
+
+#if CPU(ARM64)
+WTF_EXPORT_PRIVATE const float* findFloatAlignedImpl(const float* pointer, float target, size_t length);
+
+ALWAYS_INLINE const float* findFloat(const float* pointer, float target, size_t length)
```

Comment:
I wonder whether clients in WebKit will more often want the real `findFloat` rather than the bitwise `==` version of `findFloat`. For hash tables, we found it was more useful to have bitwise comparison of floating point numbers for hash table keys rather than floating point equality. Maybe for some uses either would be OK, in which case we want the faster of the two. In other cases, it might be inconvenient that you can’t find a `NaN`.

Reply:
This function is currently aligned to JS's indexOf behavior, and it requires floating point comparison, which includes `0 == -0` => `true`. And since we can just use `find32` if we would like to have bitwise-compared version, I think this `findFloat` is nice.